ports/blake3/CMakeLists.txt

```
@@ -13,17 +13,40 @@ target_sources(blake3 PRIVATE
 # This is a bit too exotic to be worth supporting right now
 target_compile_definitions(blake3 PRIVATE BLAKE3_NO_AVX512=1)
 
-if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
     target_sources(blake3 PRIVATE
         c/blake3_avx2.c
         c/blake3_sse2.c
         c/blake3_sse41.c
     )
-    if (NOT MSVC)
+    if(NOT MSVC)
         set_source_files_properties(c/blake3_avx2.c COMPILE_FLAGS -mavx2)
         set_source_files_properties(c/blake3_sse2.c COMPILE_FLAGS -msse2)
         set_source_files_properties(c/blake3_sse41.c COMPILE_FLAGS -msse4.1)
     endif()
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    if(WIN32 AND MSVC)
+        enable_language(ASM_MASM)
+        target_sources(blake3 PRIVATE
+            c/blake3_avx2_x86-64_windows_msvc.asm
+            c/blake3_sse2_x86-64_windows_msvc.asm
+            c/blake3_sse41_x86-64_windows_msvc.asm

```

```suggestion
            c/blake3_sse41_x86-64_windows_msvc.asm
            c/blake3_avx512_x86-64_windows_msvc.asm      
```
Why not add files supporting `AVX-512` instruction set? :)
https://github.com/BLAKE3-team/BLAKE3/blob/1.3.1/c/blake3_avx512_x86-64_windows_msvc.asm

---

Sure, I can add it. Was reluctant to bother with it because I have no data on performance and there's not a ton of hardware out there that can run it.